From 446d5b17558a0c9104dcef2808ba00fb3efdaaa5 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 14 May 2020 20:29:58 -0700
Subject: [PATCH] CMakeLists.txt: add missing dynamic_reconfigure to
 CATKIN_DEPENDS

* fixes moveit-ros-move-group issue:
/jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/aarch64-oe-linux/moveit-ros-move-group/1.0.2-1-r0/recipe-sysroot-native/usr/bin/aarch64-oe-linux/../../libexec/aarch64-oe-linux/gcc/aarch64-oe-linux/8.2.0/ld: warning: libdynamic_reconfigure_config_init_mutex.so, needed by /jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/aarch64-oe-linux/moveit-ros-move-group/1.0.2-1-r0/recipe-sysroot/opt/ros/melodic/lib/libmoveit_planning_scene_monitor.so, not found (try using -rpath or -rpath-link)
/jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/aarch64-oe-linux/moveit-ros-move-group/1.0.2-1-r0/recipe-sysroot-native/usr/bin/aarch64-oe-linux/../../libexec/aarch64-oe-linux/gcc/aarch64-oe-linux/8.2.0/ld: /jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/aarch64-oe-linux/moveit-ros-move-group/1.0.2-1-r0/recipe-sysroot/opt/ros/melodic/lib/libmoveit_planning_scene_monitor.so: undefined reference to `dynamic_reconfigure::__init_mutex__'
collect2: error: ld returned 1 exit status

* and moveit-ros-warehouse issue:
/jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/core2-64-oe-linux/moveit-ros-warehouse/1.0.2-1-r0/recipe-sysroot-native/usr/bin/x86_64-oe-linux/../../libexec/x86_64-oe-linux/gcc/x86_64-oe-linux/8.2.0/ld: warning: libdynamic_reconfigure_config_init_mutex.so, needed by /jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/core2-64-oe-linux/moveit-ros-warehouse/1.0.2-1-r0/recipe-sysroot/opt/ros/melodic/lib/libmoveit_planning_scene_monitor.so, not found (try using -rpath or -rpath-link)
/jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/core2-64-oe-linux/moveit-ros-warehouse/1.0.2-1-r0/recipe-sysroot-native/usr/bin/x86_64-oe-linux/../../libexec/x86_64-oe-linux/gcc/x86_64-oe-linux/8.2.0/ld: /jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/core2-64-oe-linux/moveit-ros-warehouse/1.0.2-1-r0/recipe-sysroot/opt/ros/melodic/lib/libmoveit_planning_scene_monitor.so: undefined reference to `dynamic_reconfigure::__init_mutex__'
collect2: error: ld returned 1 exit status

/jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/core2-64-oe-linux/moveit-ros-warehouse/1.0.2-1-r0/recipe-sysroot-native/usr/bin/x86_64-oe-linux/../../libexec/x86_64-oe-linux/gcc/x86_64-oe-linux/8.2.0/ld: warning: libdynamic_reconfigure_config_init_mutex.so, needed by /jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/core2-64-oe-linux/moveit-ros-warehouse/1.0.2-1-r0/recipe-sysroot/opt/ros/melodic/lib/libmoveit_planning_scene_monitor.so, not found (try using -rpath or -rpath-link)
/jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/core2-64-oe-linux/moveit-ros-warehouse/1.0.2-1-r0/recipe-sysroot-native/usr/bin/x86_64-oe-linux/../../libexec/x86_64-oe-linux/gcc/x86_64-oe-linux/8.2.0/ld: /jenkins/mjansa/build/ros/ros1-melodic-thud/tmp-glibc/work/core2-64-oe-linux/moveit-ros-warehouse/1.0.2-1-r0/recipe-sysroot/opt/ros/melodic/lib/libmoveit_planning_scene_monitor.so: undefined reference to `dynamic_reconfigure::__init_mutex__'
collect2: error: ld returned 1 exit status

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ff06cddc..747e9fa0d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,6 +67,7 @@ catkin_package(
     moveit_msgs
     tf2_msgs
     tf2_geometry_msgs
+    dynamic_reconfigure
   DEPENDS
     EIGEN3
 )
